src/Microsoft.VisualStudio.Threading.Analyzers.Tests/Helpers/DiagnosticVerifier.Helper.cs

```diff
@@ -1,279 +0,0 @@
-        /// </summary>
-        /// <param name="source">Classes in the form of a string</param>
-        /// <param name="language">The language the source code is in</param>
-        /// <param name="hasEntrypoint"><c>true</c> to set the compiler in a mode as if it were compiling an exe (as opposed to a dll).</param>
-        /// <returns>A Document created from the source string</returns>
-        protected static Document CreateDocument(string source, string language = LanguageNames.CSharp, bool hasEntrypoint = false)
-        {
-            return CreateProject(new[] { source }, language, hasEntrypoint).Documents.First();
-        }
-
-        /// <summary>
-        /// Create a project using the inputted strings as sources.
-        /// </summary>
-        /// <param name="sources">Classes in the form of strings</param>
-        /// <param name="language">The language the source code is in</param>
-        /// <param name="hasEntrypoint"><c>true</c> to set the compiler in a mode as if it were compiling an exe (as opposed to a dll).</param>
-        /// <returns>A Project created out of the Documents created from the source strings</returns>
-        protected static Project CreateProject(string[] sources, string language = LanguageNames.CSharp, bool hasEntrypoint = false)
-        {
-            string fileNamePrefix = DefaultFilePathPrefix;
-            string fileExt = language == LanguageNames.CSharp ? CSharpDefaultFileExt : VisualBasicDefaultExt;
-
-            var projectId = ProjectId.CreateNewId(debugName: TestProjectName);
-
-            var solution = new AdhocWorkspace()
-                .CurrentSolution
-                .AddProject(projectId, TestProjectName, TestProjectName, language)
-                .AddMetadataReference(projectId, CorlibReference)
-                .AddMetadataReference(projectId, SystemReference)
-                .AddMetadataReference(projectId, SystemCoreReference)
-                .AddMetadataReference(projectId, CSharpSymbolsReference)
-                .AddMetadataReference(projectId, CodeAnalysisReference)
-                .AddMetadataReference(projectId, ThreadingReference)
-                .AddMetadataReference(projectId, WindowsBaseReference)
-                .AddMetadataReference(projectId, OleInteropReference)
-                .WithProjectCompilationOptions(
-                    projectId,
-                    new CSharpCompilationOptions(
-                        hasEntrypoint ? OutputKind.ConsoleApplication : OutputKind.DynamicallyLinkedLibrary,
-                        specificDiagnosticOptions: new Dictionary<string, ReportDiagnostic>
-                        {
-                            { "CS1701", ReportDiagnostic.Suppress }, // we don't reference mscorlib, which can cause assembly reference ambiguities
-                        }))
-                .WithProjectParseOptions(projectId, new CSharpParseOptions(LanguageVersion.Latest));
```

Comment:
📝 The library is now using `LanguageVersion.Default`, but one test required C# 7.1 so it was configured at the test site.

Reply:
Why not just use `Latest` everywhere?

---
In reply to: [213854985](https://github.com/Microsoft/vs-threading/pull/359#discussion_r213854985) [](ancestors = 213854985)

Reply:
I tend to avoid `Latest` since I don't like the uncertainty of it. But it could be used.

Reply:
I prefer deterministic behavior as well. But isn't `Default` and `Latest` *both* subject to their values changing? And anyway, neither will ever change except when we're explicitly upgrading the version of Roslyn that we reference in our package references, I think. At that point, we're expecting things to move around a bit and to possibly fix up stuff.